pkg/admin/prerun/prerun.go

```diff
@@ -45,12 +47,14 @@ func New(dataManager data.Manager) *PreRun {
 func (pr *PreRun) Perform() error {
 	klog.InfoS("Starting pre-run")
 
+	if continuePrerun, err := pr.checkPreconditions(); err != nil {
```

Comment:
It would be good to log before and after this call, since it's a distinct unit of work.